src/main/java/org/apache/commons/dbcp2/PoolableConnection.java

```diff
@@ -124,6 +124,9 @@ public PoolableConnection(final Connection conn, final ObjectPool<PoolableConnec
     protected void passivate() throws SQLException {
         super.passivate();
         setClosedInternal(true);
+        if (getDelegateInternal() instanceof PoolingConnection) {
+            ((PoolingConnection) getDelegateInternal()).connectionReturnedToPool();
+        }
     }
```

Comment:
This is a little awkward.  I see now how my initial idea does not work.  Implementing passivate in PoolingConnection would do nothing as it would never be called.  I was tempted to recommend that we change PoolableConnection passivate to cascade to its delegate, but the problem is the delegate could just be a Connection (physical).  So no way around this that I can see.